### PR TITLE
[ISSUE #4864] Remove artifacts with incompatible licenses from jdbc-connector

### LINKED ISSUE #4864: [Enhancement] Remove artifacts with incompatible licenses from jdbc-connector
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/eventmesh/issues?q=is%3Aissue) and found no similar issues.

### Enhancement Request

Remove artifacts with incompatible licenses from jdbc-connector

- #4842 

### Describe the solution you'd like

Remove artifacts with incompatible licenses from jdbc-connector

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct) *

### PR DESCRIPTION
**Pull request by mxsm**
This PR is a subtask of #4842

fix #4864 

### Motivation

*Explain the content here.*
*Explain why you want to make the changes and what problem you're trying to solve.*

### Modifications

*Describe the modifications you've done.*

### Documentation

- Does this pull request introduce a new feature? (yes / no)
- If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)
- If a feature is not applicable for documentation, explain why?
- If a feature is not documented yet in this PR, please create a followup issue for adding the documentation

### CODE REVIEW

**Comment by Pil0tXia** [comment]
Please be informed that the PR title doesn't correspond to its change.

File: eventmesh-connectors/eventmesh-connector-jdbc/build.gradle [hunk 1]
```diff
@@ -37,12 +37,12 @@ dependencies {
...
     implementation 'org.antlr:antlr4-runtime:4.13.0'
     implementation 'com.alibaba:druid:1.2.20'
-    implementation 'org.hibernate:hibernate-core:5.6.15.Final'
+    compileOnly 'org.hibernate:hibernate-core:5.6.15.Final'
     implementation project(":eventmesh-common")
     implementation project(":eventmesh-openconnect:eventmesh-openconnect-java")
     implementation project(":eventmesh-spi")
     implementation 'com.zendesk:mysql-binlog-connector-java:0.28.0'
-    implementation 'mysql:mysql-connector-java:8.0.32'
+    compileOnly 'mysql:mysql-connector-java:8.0.32'
```
Comment (Pil0tXia) [inline eventmesh-connectors/eventmesh-connector-jdbc/build.gradle:46]: `mysql-connector-java` has been redirected to `mysql-connector-j`.

Although switching to `compileOnly` scope, its artifact jar package will still be bundled in our binary distribution. Therefore we will have to remove it manually before releasing, inform users to download its jar package, and put the jar file into the `lib` directory.

**Comment by mxsm** [comment]
> Please be informed that the PR title doesn't correspond to its change.

Please review this PR try again @Pil0tXia

File: eventmesh-connectors/eventmesh-connector-jdbc/build.gradle (hunk 1 above)
Comment (Pil0tXia) [reply eventmesh-connectors/eventmesh-connector-jdbc/build.gradle:46]: This conversation is a inform and no changes are requested.

**Comment by Pil0tXia** [comment]
Since #4842 contains another artifact `xsdlib`, may you please create another issue to be closed? I will add this issue into #4842~

**Comment by mxsm** [comment]
The adjustment no longer requires deletion during packaging, as the package will not be included in the build, as shown in the figure below.
image

**Review by Pil0tXia — DISMISSED** [review]

File: eventmesh-connectors/eventmesh-connector-jdbc/build.gradle (hunk 1 above)
Comment (Pil0tXia) [reply eventmesh-connectors/eventmesh-connector-jdbc/build.gradle:46]: Sorry I mistook it as `runtimeOnly`. It do remove artifacts from classpath, and no more manual operation is needed. Users will still have to download and put jar file by themselves.

**Comment by Pil0tXia** [comment]
I just found that `ST4` is an incompatible artifact:

image

File: tools/dependency-check/known-dependencies.txt [hunk 2]
```diff
@@ -1,4 +1,3 @@
-FastInfoset-1.2.15.jar
 ST4-4.3.4.jar
 accessors-smart-2.4.7.jar
```
Comment (Pil0tXia) [inline tools/dependency-check/known-dependencies.txt:2]: According to https://github.com/apache/eventmesh/pull/4863#issuecomment-2076542000, shall we remove `ST4` together in this PR?

Comment (mxsm) [reply tools/dependency-check/known-dependencies.txt:2]: After rebasing the code, I used a plugin to check and found no errors. The status is passed

image

Comment (Pil0tXia) [reply tools/dependency-check/known-dependencies.txt:2]: The license content of https://www.antlr.org/license.html is identical with BSD-3-Clause. It seems to be an error of upstream supply chain, and `ST4` is compatible.

**Review by Pil0tXia — APPROVED** [review]
LGTM

**Review by xwm1992 — APPROVED** [review]

**Pull request merged**